test/test_helper.rb

```diff
@@ -1,2 +1,6 @@
 require 'minitest/autorun'
 require 'liquid/c'
+
+if GC.respond_to?(:compact)
+  GC.compact
+end
```

Comment:
Instead of doing this here, what about doing it after each test?

Reply:
I guess that would be even more robust. But I do expect most objects we hold onto from the C side to be moved after the first call.

If you think repeated calls would be better an can do that.

Reply:
I think it's worth it because it could save us from future compaction bugs.

Reply:
Make sense. Done.